docs/googlecloud.rst

```diff
@@ -0,0 +1,219 @@
+
+* Go to APIs & Services → Credentials →  Create Credentials
+* Select Service account key.
+* Download the json file and save as "creds.json".
+
+Export as env variable::
+
+    export GOOGLE_APPLICATION_CREDENTIALS=$PWD/creds.json
+
+User & SSH key management
+-------------------------
+
+By default Nextflow creates in each GCE instance a user with the same name as the one in your local computer and install the SSH public key available at the path $HOME/.ssh/id_rsa.pub. A different user/key can be specified as shown below::
+
+    cloud {
+        userName = 'the-user-name'
+        keyFile = '/path/to/ssh/key.pub'
+        driver = 'gce'
+    }
+
+.. note:: The configuration of your instance on GCP requires a driver parameter as shown in the example above.
+
+
+Cluster deployment
+------------------
+
+Once you have defined the configuration settings in the ``nextflow.config`` file you can create the cloud cluster by using the following command::
+
+    ./nextflow cloud create my-cluster -c <num-of-nodes>
+
+The string ``my-cluster`` identifies the cluster instance. Replace it with a name of your choice.
+
+Finally replace ``<num-of-nodes>`` with the actual number of instances that will make-up the cluster. One node is created as master, the remaining as workers. If the option ``-c`` is omitted only the **master** node
+is created.
+
+.. warning:: You will be charged accordingly the type and the number of instances chosen.
+
+The console will display the configuration that you have defined and ask you to confirm creation of the cluster with the configuration displayed. It will take some time for the cluster to deploy. You should be able to track the status of the deployed in the administration console for the Google Cloud Platform under VM instances in the Compute Engine section.
+
+
+Pipeline execution
+==================
+
+Once the cluster initialisation is complete, Nextflow will display the SSH command to connect to the master node. Use that command to connect to the cluster.
+
+.. note:: On MacOS, use the following command to avoid being asked for a pass-phrase even
+  you haven't defined one::
+
+    ssh-add -K [private key file]
+
+The suggested approach is to run your pipeline downloading it from a public repository such as GitHub and to pack the binaries dependencies in a Docker container as described in the :ref:`Pipeline sharing <sharing-page>` section.
+
+Cluster shutdown
+================
+
+When completed shutdown the cluster instances by using the following command::
+
+    nextflow cloud shutdown my-cluster
+
+Preemptible instances 
+=====================
+
+An optional parameter allows you to set the instance to be preemptible. Both master and worker instances can be set to be preemptible. The following example shows a cluster configuration with a preemptible setting::
+
+    cloud {
+        imageId = ‘centos-cloud/global/images/centos-7-v20181011’
+        instanceType = ‘n1-highcpu-8’
```

Comment:
Please replace curly quotes `‘` with `'` otherwise copying & pasting will be in trouble.

Reply:
Changed.  Will push the changes later today.